src/lxc/console.c

```diff
@@ -683,6 +683,7 @@ int lxc_console(struct lxc_container *c, int ttynum,
 	ts->escape = escape;
 	ts->winch_proxy = c->name;
 	ts->winch_proxy_lxcpath = c->config_path;
+	ts->stdoutfd = stdoutfd;
```

Comment:
ts->stdinfd had been filled in the method `lxc_console_sigwinch_init`
```
struct lxc_tty_state *lxc_console_sigwinch_init(int srcfd, int dstfd)
{
	sigset_t mask;
	struct lxc_tty_state *ts;

	ts = malloc(sizeof(*ts));
	if (!ts)
		return NULL;

	memset(ts, 0, sizeof(*ts));
	ts->stdinfd = srcfd;
	ts->masterfd = dstfd;
	ts->sigfd = -1;
        .......

```
The ts->stderrfd did not use at all. becase the stdin\stdout\stderr of container had been duplicated to the same slave of pty . So the stdout and stderr of container are combined. 
```
int lxc_console_set_stdfds(int fd)
{
    if (fd < 0)
        return 0;

    if (isatty(STDIN_FILENO))
        if (dup2(fd, STDIN_FILENO) < 0) {
            SYSERROR("failed to duplicate stdin.");
            return -1;
        }

    if (isatty(STDOUT_FILENO))
        if (dup2(fd, STDOUT_FILENO) < 0) {
            SYSERROR("failed to duplicate stdout.");
            return -1;
        }

    if (isatty(STDERR_FILENO))
        if (dup2(fd, STDERR_FILENO) < 0) {
            SYSERROR("failed to duplicate stderr.");
            return -1;
        }

    return 0;
} 
```
So, we can get the stdout\stderr info of container by reading the master of pty (ts->masterfd) . And then we just need to write the info to the ts->stdout, the info include the 'stdout' and 'stderr'.
```
int lxc_console_cb_tty_master(int fd, uint32_t events, void *cbdata,
		struct lxc_epoll_descr *descr)
{
	struct lxc_tty_state *ts = cbdata;
	char buf[1024];
	int r, w;

	if (fd != ts->masterfd)
		return 1;

	r = lxc_read_nointr(fd, buf, sizeof(buf));
	if (r <= 0)
		return 1;

	w = lxc_write_nointr(ts->stdoutfd, buf, r);
	if (w <= 0) {
		return 1;
	} else if (w != r) {
		SYSERROR("failed to write");
		return 1;
	}

	return 0;
}
```